src/runtime/composables/local/useAuth.ts

```diff
@@ -135,27 +157,85 @@ const signUp = async (credentials: Credentials, signInOptions?: SecondarySignInO
   return signIn(credentials, signInOptions)
 }
 
+const refresh = async (getSessionOptions?: GetSessionOptions) => {
+  const nuxt = useNuxtApp()
+  const config = useTypedBackendConfig(useRuntimeConfig(), 'local')
+
+  // Only refresh the session if the refresh logic is not enabled
+  if (!config.refresh.isEnabled) {
+    return getSession(getSessionOptions)
+  }
+
+  const { path, method } = config.refresh.endpoint
+  const refreshRequestTokenPointer = config.refresh.token.refreshRequestTokenPointer
+
+  const { refreshToken, token, rawToken, rawRefreshToken, lastRefreshedAt } = useAuthState()
+
+  const headers = new Headers({
+    [config.token.headerName]: token.value
+  } as HeadersInit)
+
+  const response = await _fetch<Record<string, any>>(nuxt, path, {
+    method,
+    headers,
+    body: objectFromJsonPointer(refreshRequestTokenPointer, refreshToken.value)
+  })
+
+  // Extract the new token from the refresh response
+  const extractedToken = jsonPointerGet(response, config.token.signInResponseTokenPointer)
+  if (typeof extractedToken !== 'string') {
+    console.error(
+      `Auth: string token expected, received instead: ${JSON.stringify(extractedToken)}.
+       Tried to find token at ${config.token.signInResponseTokenPointer} in ${JSON.stringify(response)}`
+    )
+    return
+  }
+
+  if (!config.refresh.refreshOnlyToken) {
+    const refreshTokenPointer = config.refresh.token.signInResponseRefreshTokenPointer
+    const extractedRefreshToken = jsonPointerGet(response, refreshTokenPointer)
+    if (typeof extractedRefreshToken !== 'string') {
+      console.error(
+        `Auth: string token expected, received instead: ${JSON.stringify(extractedRefreshToken)}.
+         Tried to find refresh token at ${refreshTokenPointer} in ${JSON.stringify(response)}`
+      )
+      return
+    } else {
+      rawRefreshToken.value = extractedRefreshToken
+    }
+  }
+
+  rawToken.value = extractedToken
+  lastRefreshedAt.value = new Date()
+
+  await nextTick()
+  return getSession(getSessionOptions)
+}
+
 interface UseAuthReturn extends CommonUseAuthReturn<typeof signIn, typeof signOut, typeof getSession, SessionData> {
   signUp: typeof signUp
   token: Readonly<Ref<string | null>>
+  refreshToken: Readonly<Ref<string | null>>
```

Comment:
Could you please leave a JSDoc here to signify that this will always be `null` when `refresh.isEnabled` set to `false`?

Reply:
I used a `@remark` field for this! I hope this is correct 😊 

<img width="741" alt="Screenshot 2024-08-09 at 14 16 39" src="https://github.com/user-attachments/assets/9c456628-0b66-446c-af17-caba67ff95d7">